service/src/test/java/bio/terra/workspace/pact/TpsApiTest.java

```diff
@@ -0,0 +1,555 @@
+package bio.terra.workspace.pact;
+
+import static au.com.dius.pact.consumer.dsl.DslPart.UUID_REGEX;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+import au.com.dius.pact.consumer.MockServer;
+import au.com.dius.pact.consumer.dsl.PactDslJsonArray;
+import au.com.dius.pact.consumer.dsl.PactDslJsonBody;
+import au.com.dius.pact.consumer.dsl.PactDslWithProvider;
+import au.com.dius.pact.consumer.junit5.PactConsumerTestExt;
+import au.com.dius.pact.consumer.junit5.PactTestFor;
+import au.com.dius.pact.core.model.RequestResponsePact;
+import au.com.dius.pact.core.model.annotations.Pact;
+import bio.terra.policy.model.*;
+import bio.terra.workspace.app.configuration.external.FeatureConfiguration;
+import bio.terra.workspace.app.configuration.external.PolicyServiceConfiguration;
+import bio.terra.workspace.service.iam.AuthenticatedUserRequest;
+import bio.terra.workspace.service.policy.TpsApiDispatch;
+import bio.terra.workspace.service.policy.exception.PolicyServiceNotFoundException;
+import bio.terra.workspace.service.resource.exception.PolicyConflictException;
+import bio.terra.workspace.service.workspace.WorkspaceService;
+import bio.terra.workspace.service.workspace.model.CloudPlatform;
+import bio.terra.workspace.service.workspace.model.Workspace;
+import bio.terra.workspace.service.workspace.model.WorkspaceStage;
+import java.util.*;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.http.HttpStatus;
+
+@Tag("pact-test")
+@ExtendWith(PactConsumerTestExt.class)
+public class TpsApiTest {
+
+  // Note: the header must match exactly so pact doesn't add it's own
+  // if "Content-type" is specified instead,
+  // pact will also have a required header for "Content-Type: application/json; charset=UTF-8"
+  // which will cause the request to fail to match,
+  // since our client doesn't include the encoding in the content type header
+  static Map<String, String> contentTypeJsonHeader = Map.of("Content-Type", "application/json");
+
+  // The policy ids aren't significant - hardcoded instead of random to avoid changing the pact on
+  // every run
+  static UUID existingPolicyId = UUID.fromString("bea1edd0-a8e6-4d60-a613-e8e065f21616");
+  static String existingPolicyState = "an existing policy";
+  static String existingPolicyProviderStateValue = "${policyId}";
+  static UUID secondPolicyId = UUID.fromString("a254714b-4519-4ce4-ad87-19a7143376f4");
+  static String secondPolicyState = "a second existing policy";
```

Comment:
I don't have a better name. The details of what the policy are aren't important here: just any policy, in order to test linking them.
When the provider side pacts are written, maybe someone will come up with better names, and we can go back and update this.